plugins/embedding.cc

```diff
@@ -286,6 +286,37 @@ struct EmbeddingDataset::Itl
         return result;
     }
 
+    struct EmbeddingRowStream : public RowStream {
+
+        EmbeddingRowStream(const EmbeddingDataset::Itl* source) : index(0), source(source)
+        {
+         
+        }
+
+        virtual std::shared_ptr<RowStream> clone() const{
```

Comment:
We need to clone the stream because in the boundquery each task needs its own. If / when we do the "splitIntoManyStreams" we shouldnt need the clone anymore. Im commenting the methods in dataset.h